content/geometry/PolygonUnionArea.h

```diff
@@ -0,0 +1,49 @@
+/**
+ * Author: black_horse2014, chilli
+ * Date: 2019-10-29
+ * License: CC0
+ * Source:
+ * Description: Calculates the area of the union of $n$ polygons (not necessarily convex).
+ * Status: Submitted on ECNA 2017 Problem A.
+ * Time: $O(N^2)$, where $N$ is the total number of points
+ * Usage:
+ */
+#pragma once
+
+#include "Point.h"
+#include "sideOf.h"
+
+typedef Point<double> P;
+double rat2(P a, P b) { return sgn(b.x) ? a.x/b.x : a.y/b.y; }
+double polyUnion(vector<vector<P>> &poly) {
+	double ret = 0;
+	rep(i,0,sz(poly)) rep(v,0,sz(poly[i])) {
+		P A = poly[i][v], B = poly[i][(v + 1) % sz(poly[i])];
+		vector<pair<double, int>> segs({{0, 0}, {1, 0}});
+		rep(j,0,sz(poly)) {
+			if (i == j) continue;
+			rep(u,0,sz(poly[j])) {
+				P C = poly[j][u], D = poly[j][(u + 1) % sz(poly[j])];
```

Comment:
There's a lot of `poly[i]` and `poly[j]` and `poly` here, which are long and easily confusable. How about `auto &A = poly[i], &B = poly[j]`? (... well, ok, with other names because you're already using A and B)

Reply:
Unfortunately, that does add 2 lines :/, since the `poly[i]` and `poly[j]` are used in different places.